### DISCUSSION TITLE
v7.2.1 fails to work with CFL numbers that v7.0.6 was able to work fine

### DISCUSSION THREAD

**Discussion by MKursatUzuner** [su2code/SU2#1522] [Q&A] [answered]
SU2 v7.2.1 fails to work at CFL numbers that v7.0.6 was able to work fine. 

I have been using v7.0.6 and wanted to try newer versions and started working with v7.2.1. However, I experienced something I was not expecting that is v7.2.1 would diverge at the CFL numbers v7.0.6 was able to run. I experienced this behavior in several cases.

For a 3D turbomachinery case, v7.0.6 was able to run at CFL 20 while v7.2.1 was only able to run at CFL 1.
For a 3D external flow case, v7.0.6 was able to run at CFL 100 while v7.2.1 was only able to run at CFL 10.
I didn't do a result comparison for these cases.

To reproduce the same behavior I have also used 2D rae2822 airfoil case. For this, v7.0.6 was able to run at CFL 1000 while v7.2.1 was able to run at CFL 250. When results are compared v7.2.1 with CFL 250 was less accurate compared to v7.0.6 with CFL 1000 as well. I have shared the pressure coefficient comparison between v7.0.6, v7.2.1 and the experiment below.

![image](https://user-images.githubusercontent.com/66836581/150640337-3b1439c1-85d2-481c-9188-720f9f0958a9.png)

I have included the .cfg files I have used and code outputs for v7.0.6 and v7.2.1.

[codeOutput_v706.txt](https://github.com/su2code/SU2/files/7918433/codeOutput_v706.txt)
[codeOutput_v721.txt](https://github.com/su2code/SU2/files/7918436/codeOutput_v721.txt)

[rae2822cfg_v706.txt](https://github.com/su2code/SU2/files/7918444/rae2822cfg_v706.txt)
[rae2822cfg_v721.txt](https://github.com/su2code/SU2/files/7918445/rae2822cfg_v721.txt)

**Comment by pcarruscag** [comment]
Hi the same issue is reported here: https://www.cfd-online.com/Forums/su2/240690-stable-solution-version-7-0-0-diverging-7-2-0-a.html
I think it is because of a bug we fixed that could cause the solution in some places to either freeze or be updated very slowly, seeing that the residuals in your 7.0.6 example don't drop, this is likely to be happening.
If you share the mesh you are using I can try suggesting better settings, to get to high CFL number you usually need CFL adaptation.

**Comment by MKursatUzuner** [comment]
I tried using adaptive CFL as you suggested for v7.2.1 with settings below;

CFL_ADAPT_PARAM= ( 0.1, 2.0, 1.0, 1e10, 0.001 )

Even though it doesn't diverge it still needs more iterations to converge compared to v7.0.6 and result is pretty much the same with constant CFL 250. Also, with adaptive CFL residuals are changing but they are oscillating in an interval. 

What I don't really understand is why using same settings would yield less accurate results in a newer version.

Since these settings are working with v7.0.6 and are very common I don't think I need to alter them for a newer version to give the same results.

Here is the code output for v7.2.1 with adaptive cfl enabled.

[codeOutput_v721_CFLadapt.txt](https://github.com/su2code/SU2/files/7919208/codeOutput_v721_CFLadapt.txt)

**Comment by pcarruscag** [comment]
If you share the mesh I should be able to give better guidance.

**Comment by MKursatUzuner** [comment]
Okay, here is the mesh in su2 format.

-

**Comment by pcarruscag** [comment, marked as answer]
Thanks for that. The boundary conditions you had are not ideal, the inlet flow was directed against the upper slip wall.
That type of mesh is also not the best, you should try to keep the first layers normal to the surface, and the wake of very stretched elements is also an issue, if possible try to "open" those elements as they move away from the surface to avoid the extremely high aspect ratios away from the surface, it slows down the convergence of the code significantly, and it also makes multigrid impossible.
With the settings below the residuals drops enough to have some confidence in the results.
[config.txt](https://github.com/su2code/SU2/files/7921349/config.txt)
I think everything is ok, the convergence was just being saved by a bug we had.
![image](https://user-images.githubusercontent.com/38071223/150696994-e4feeb74-ad7b-413d-b5ec-b405b9c0d065.png)

**Reply by pcarruscag** [reply]
namely #1036 introduced in version 7.1.0
https://github.com/su2code/SU2/releases/tag/v7.1.0

**Reply by MKursatUzuner** [reply]
Thanks for your time and answers.
